examples/lyapunov.rs

```diff
@@ -0,0 +1,46 @@
+//! Lyapunov Analysis example
+
+extern crate ndarray;
+extern crate ndarray_odeint;
+extern crate ndarray_linalg;
+extern crate num_traits;
+
+use ndarray::*;
+use ndarray_linalg::*;
+use ndarray_odeint::*;
+use num_traits::Float;
+
+/// Calculate all Lyapunov exponents
+pub fn exponents<A, TEO>(teo: TEO, x0: Array1<A>, alpha: A::Real, duration: usize) -> Array1<A>
+    where A: RealScalar,
+          TEO: TimeStep<A::Real>,
+          for<'a> &'a TEO: TimeEvolution<OwnedRepr<A>, Ix1>,
+          for<'a, 'b> Jacobian<'a, A, OwnedRepr<A>, Ix1, TEO>: OperatorMut<ViewRepr<&'b mut A>, Ix1>
+{
+    let n = x0.len();
+    let dur: A = AssociatedReal::inject(teo.get_dt() * into_scalar(duration as f64));
+    let ts = time_series(x0, &teo);
+    ts.scan(Array::eye(n), |q: &mut Array2<A>, x| {
+        let j = jacobian(&teo, x.clone(), alpha);
+        let q = j.op_multi_mut(q);
+        let (q_next, r) = q.qr().unwrap();
+        *q = q_next;
+        let d: Array1<A> = r.diag().map(|x| AssociatedReal::inject(x.abs().ln()));
```

Comment:
`ln` does not exist for `Scalar`
https://github.com/termoshtt/ndarray-linalg/issues/63